src/hmem_rocr.c

```diff
@@ -200,12 +225,30 @@ static int rocr_host_memory_ptr(void *host_ptr, void **ptr)
 		return -FI_EIO;
 	}
 
-	if (info.type != HSA_EXT_POINTER_TYPE_LOCKED)
-		*ptr = host_ptr;
-	else
+	if (agent)
+		*agent = info.agentOwner;
+
+	if (size)
+		*size = info.sizeInBytes;
+
+	if (info.type != HSA_EXT_POINTER_TYPE_LOCKED) {
+		if (info.type == HSA_EXT_POINTER_TYPE_IPC ||
+			info.type == HSA_EXT_POINTER_TYPE_HSA)
+			*ptr = info.agentBaseAddress;
+		else
+			*ptr = host_ptr;
+
+		if (info.type == HSA_EXT_POINTER_TYPE_UNKNOWN && system)
+			*system = true;
+		if (offset)
+			*offset = host_ptr - *ptr;
+	} else {
 		*ptr = (void *) ((uintptr_t) info.agentBaseAddress +
 				 (uintptr_t) host_ptr -
 				 (uintptr_t) info.hostBaseAddress);
+		if (system)
+			*system = true;
```

Comment:
This was already set above

Reply:
not sure I follow. system is initialized to false and set to true under two conditions: memory type is Unknown or locked.

Reply:
I misread the top assignment